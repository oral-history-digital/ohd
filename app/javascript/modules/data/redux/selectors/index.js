export * from './baseSelectors';
export * from './dataSelectors';
