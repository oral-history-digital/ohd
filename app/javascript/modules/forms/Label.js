import PropTypes from 'prop-types';
import classNames from 'classnames';

import { useI18n } from 'modules/i18n';

function Label({
    label,
    labelKey,
    mandatory,
    htmlFor,
    className,
}) {
    const { t } = useI18n();

    const l = label ? label : t(labelKey);

    return (
        <div className={classNames('form-label', className)}>
            <label className="FormLabel" htmlFor={htmlFor}>
                {`${l}${mandatory ? ' *' : ''}`}
            </label>
        </div>
    );
}

Label.propTypes = {
    label: PropTypes.string,
    labelKey: PropTypes.string,
    mandatory: PropTypes.bool.isRequired,
    htmlFor: PropTypes.string,
    className: PropTypes.string,
};

Label.defaultProps = {
    mandatory: false,
};

export default Label;
