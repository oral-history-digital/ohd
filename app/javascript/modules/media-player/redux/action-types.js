import { NAME } from '../constants';

export const UPDATE_MEDIA_TIME = `${NAME}/updateMediaTime`;
export const UPDATE_IS_PLAYING = `${NAME}/updateIsPlaying`;
export const SET_TAPE = `${NAME}/setTape`;
export const RESET_MEDIA = `${NAME}/resetMedia`;
export const SEND_TIME_CHANGE_REQUEST = `${NAME}/sendTimeChangeRequest`;
export const CLEAR_TIME_CHANGE_REQUEST = `${NAME}/clearTimeChangeRequest`;
export const SET_PLAYER_SIZE = `${NAME}/setPlayerSize`;
