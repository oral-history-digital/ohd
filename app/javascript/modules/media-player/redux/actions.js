import {
    UPDATE_MEDIA_TIME,
    UPDATE_IS_PLAYING,
    SET_TAPE,
    RESET_MEDIA,
    SEND_TIME_CHANGE_REQUEST,
    CLEAR_TIME_CHANGE_REQUEST,
    SET_PLAYER_SIZE,
} from './action-types';

export const updateMediaTime = (time) => ({
    type: UPDATE_MEDIA_TIME,
    payload: { time },
});

export const updateIsPlaying = (isPlaying) => ({
    type: UPDATE_IS_PLAYING,
    payload: { isPlaying },
});

export const setTape = (tape) => ({
    type: SET_TAPE,
    payload: { tape },
});

export const resetMedia = () => ({ type: RESET_MEDIA });

export const sendTimeChangeRequest = (tape, time) => ({
    type: SEND_TIME_CHANGE_REQUEST,
    payload: {
        tape,
        time,
    },
});

export const clearTimeChangeRequest = () => ({ type: CLEAR_TIME_CHANGE_REQUEST });

export const setPlayerSize = (size) => ({
    type: SET_PLAYER_SIZE,
    payload: { size },
});
