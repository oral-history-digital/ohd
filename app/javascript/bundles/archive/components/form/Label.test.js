import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import Label from './Label';

const store = createStore(f => f, {
    archive: {
        locale: 'de',
        translations: {
            de: {},
        },
    },
});

it('renders correctly', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <Label
                    label="Name"
                    className="label"
                    htmlFor="input"
                />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders with translation key', () => {
    const tree = renderer
        .create(
            <Provider store={store}>
                <Label
                    labelKey="label.default_name"
                />
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
