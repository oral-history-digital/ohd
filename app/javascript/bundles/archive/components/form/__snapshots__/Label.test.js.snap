// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-label label"
>
  <label
    className="FormLabel"
    htmlFor="input"
  >
    Name
  </label>
</div>
`;

exports[`renders with translation key 1`] = `
<div
  className="form-label"
>
  <label
    className="FormLabel"
  >
    default_name
  </label>
</div>
`;
